code/modules/projectiles/projectile/energy.dm

```diff
@@ -20,7 +24,28 @@
 	//Damage will be handled on the MOB side, to prevent window shattering.
 
 /obj/item/projectile/energy/electrode/on_hit(var/atom/target, var/blocked = 0)
+	var/deflected = NO_DEFLECTION //0 for not deflected, 1 for partially, 2 for totally.
+	if(ishuman(target)) //have to do this stuff before we call ..(), but I want the messages to come in the right order. So the deflected var.
+		var/mob/living/carbon/human/H = target//typecast
+		var/protection = H.getarmor(def_zone, "shock") //checks for taser resistance of worn armor
+		if(prob(protection)) //chance to completely negate taser hit
+			stun = 0
+			weaken = 0
+			stutter = 0
+			jitter = 0
+			deflected = FULL_DEFLECTION
+		else if(prob(protection)) //chance for a glancing blow if first check fails
+			stun = 0
+			weaken = 0
+			H.apply_effect(stamina, 50, 0) //stamina damage, still better than being stunned.
+			deflected = PARTIAL_DEFLECTION
 	. = ..()
+	if(deflected == PARTIAL_DEFLECTION)
+		target.visible_message("<span class='danger'>The shock is weakened by the armor!</span>", \
+								"<span class='userdanger'>The shock is weakened by your armor!</span>")
+	else if(deflected == FULL_DEFLECTION)
+		target.visible_message("<span class='danger'>The electrode glance off of the armor!</span>", \

```

Comment:
`The electrode glances off the armor`, because: `off of` is ugly and wrong; and electrode is singular

Reply:
Ugly is subjective, but off of is not wrong, as a quick google search will show. Still, I'll shorten it and correct the verb (It originally had the electrodes in the plural and I forgot to change the other stuff back)